components/MetricsTable.tsx

```diff
@@ -124,10 +132,30 @@ const MetricsTable = ({ metrics }: { metrics: MetricBare[] }) => {
 
   return (
     <MaterialTable
+      actions={
+        canEditMetrics
+          ? [
+              {
+                icon: 'edit',
+                tooltip: 'Edit Metric',
+                onClick: (_event, rowData) => {
+                  // istanbul ignore next; this should never happen
+                  if (!onEditMetric) {
+                    throw new Error('Missing onEditMetric while canEditMetrics = true')
+                  }
+                  onEditMetric((rowData as MetricBare).metricId)
+                },
+              },
+            ]
+          : undefined
+      }
```

Comment:
If we drop `canEditMetrics`, this will be cleaner:

```suggestion
     actions={
        onEditMetric
          ? [
              {
                icon: 'edit',
                tooltip: 'Edit Metric',
                onClick: (_event, rowData) => onEditMetric((rowData as MetricBare).metricId),
              },
            ]
          : undefined
      }
```